src/components/backgroundselector/BackgroundSelectorDirective.js

```diff
@@ -0,0 +1,138 @@
+(function() {
+  goog.provide('ga_backgroundselector_directive');
+
+  goog.require('ga_map');
+  goog.require('ga_permalink');
+
+  var module = angular.module('ga_backgroundselector_directive', [
+    'ga_map',
+    'ga_permalink',
+    'pascalprecht.translate'
+  ]);
+
+  module.directive('gaBackgroundSelector',
+    function($document, gaPermalink, gaLayers, gaLayerFilters,
+      gaBrowserSniffer) {
+      return {
+        restrict: 'A',
+        templateUrl:
+            'components/backgroundselector/partials/backgroundselector.html',
+        scope: {
+          map: '=gaBackgroundSelectorMap'
+        },
+        link: function(scope, elt, attrs) {
+          scope.isBackgroundSelectorClosed = true;
+          scope.embed = gaBrowserSniffer.embed;
+          var map = scope.map;
+          var firstLayerChangeEvent = true;
```

Comment:
ok, i'll have a try

2015-05-28 20:32 GMT+02:00 Gilbert Jeiziner <notifications@github.com>:

> In src/components/backgroundselector/BackgroundSelectorDirective.js
> <https://github.com/geoadmin/mf-geoadmin3/pull/2348#discussion_r31263654>:
>
> > +
> > +  module.directive('gaBackgroundSelector',
> > +    function($document, gaPermalink, gaLayers, gaLayerFilters,
> > +      gaBrowserSniffer) {
> > +      return {
> > +        restrict: 'A',
> > +        templateUrl:
> > +            'components/backgroundselector/partials/backgroundselector.html',
> > +        scope: {
> > +          map: '=gaBackgroundSelectorMap'
> > +        },
> > +        link: function(scope, elt, attrs) {
> > +          scope.isBackgroundSelectorClosed = true;
> > +          scope.embed = gaBrowserSniffer.embed;
> > +          var map = scope.map;
> > +          var firstLayerChangeEvent = true;
>
> yeah, but then currentTopic will be set, and the part guarded by if
> (scope.currentTopic) won't be executed again. Only the first time. The
> idea is the remove firstLayerChangeEvent variable and us currentTopic to do
> the same thing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geoadmin/mf-geoadmin3/pull/2348/files#r31263654>.
>



-- 
Twitter: http://twitter.com/cedricmoullet
Linked In: http://www.linkedin.com/in/cedricmoullet

<http://twitter.com/cedricmoullet>